src/compiler/config/validate-service-worker.ts

```diff
@@ -17,18 +37,16 @@ export const validateServiceWorker = (config: d.ValidatedConfig, outputTarget: d
     return;
   }
 
-  if (outputTarget.serviceWorker === true) {
-    outputTarget.serviceWorker = {};
-  } else if (!outputTarget.serviceWorker && config.devMode) {
+  if (!outputTarget.serviceWorker && config.devMode) {
     outputTarget.serviceWorker = null;
     return;
   }
 
-  if (typeof outputTarget.serviceWorker !== 'object') {
-    // what was passed in could have been a boolean
-    // in that case let's just turn it into an empty obj so Object.assign doesn't crash
-    outputTarget.serviceWorker = {};
-  }
+  outputTarget.serviceWorker = {
+    ...outputTarget.serviceWorker,
+    globDirectory: outputTarget.serviceWorker?.globDirectory ?? outputTarget.appDir,
+    swDest: outputTarget.serviceWorker?.swDest ?? join(outputTarget.appDir ?? '', DEFAULT_FILENAME),
```

Comment:
Do we need to validate that this is a string?
```suggestion
    swDest: isString(outputTarget.serviceWorker?.swDest) ? outputTarget.serviceWorker.swDest : join(outputTarget.appDir ?? '', DEFAULT_FILENAME),
```